### PR TITLE
Home spec: add test cases

### PR DESCRIPTION
**Pull request by lihaobhsfer**

### CODE REVIEW

File: src/plugins/sails.js [hunk 1]
```diff
@@ -1,22 +1,22 @@
-import store from '../store/store.js';
-let _ = require('lodash');
+import store from "../store/store.js";
+let _ = require("lodash");
```
Comment (itsalaidbacklife) [inline src/plugins/sails.js:2]: Any reason for the double quotes? Most strings throughout the rest of the application are single quotes unless single quotes unless a literal ' is needed etc.

File: src/plugins/sails.js [hunk 2]
```diff
@@ -25,21 +25,25 @@ io.socket.on('game', function(evData) {
 	}
 });
 
-io.socket.on('join', function(evData) {
-	console.log('Game was joined');
-	store.commit('joinGame', {
+io.socket.on("join", function(evData) {
+	console.log("Game was joined");
```
Comment (itsalaidbacklife) [inline src/plugins/sails.js:29]: Let's remove stale logs

File: src/plugins/sails.js [hunk 3]
```diff
@@ -25,21 +25,25 @@ io.socket.on('game', function(evData) {
...
 	// If we are in game: update our game with new player
 	if (evData.gameId === store.state.game.id) {
-		store.commit('opponentJoined', evData.newPlayer);
+		store.commit("opponentJoined", evData.newPlayer);
 	}
 });
 
-io.socket.on('leftGame', function(evData) {
+io.socket.on("leftGame", function(evData) {
+	console.log(evData.id, store.state.game.id);
```
Comment (itsalaidbacklife) [inline src/plugins/sails.js:42]: Stale log

File: src/plugins/sails.js [hunk 4]
```diff
@@ -25,21 +25,25 @@ io.socket.on('game', function(evData) {
...
 	if (evData.id === store.state.game.id) {
-		store.commit('opponentLeft');
+		store.commit("opponentLeft");
+	} else {
+		console.log("leftGame");
```
Comment (itsalaidbacklife) [inline src/plugins/sails.js:46]: Stale log

File: src/store/modules/auth.js [hunk 5]
```diff
@@ -59,13 +58,28 @@ export default {
...
+
+		requestLogout(context) {
+			return new Promise((resolve, reject) => {
+				io.socket.get("/user/logout", {}, function handleResponse(
+					resData,
+					jwres
+				) {
+					console.log(jwres);
+					if (jwres.statusCode === 200) {
+						console.log("logout success");
```
Comment (itsalaidbacklife) [inline src/store/modules/auth.js:77]: stale log

File: src/store/modules/auth.js [hunk 6]
```diff
@@ -59,13 +58,28 @@ export default {
 							console.log("auth success");
```
Comment (itsalaidbacklife) [inline src/store/modules/auth.js:58]: I know this was already here, but could you remove this log as well?

File: src/store/modules/auth.js [hunk 7]
```diff
@@ -50,7 +49,7 @@ export default {
 					"/user/signup",
 					{
 						email: data.email,
-						password: data.password
+						password: data.password,
 					},
 					function handleResponse(resData, jwres) {
 						console.log(resData);
```
Comment (itsalaidbacklife) [inline src/store/modules/auth.js:55]: I'm okay w/ these data logs mainly to placate the linter b/c it gives us an excuse to write the handler params consistently in advance. Open to discussion here

File: src/views/Home.vue [hunk 8]
```diff
@@ -98,6 +112,18 @@ export default {
 				.catch(() => {
 					console.log("Error creating game");
 				});
+		},
+		logout(){
+			console.log("loggin out")
```
Comment (itsalaidbacklife) [inline src/views/Home.vue:117]: Stale log

File: tests/e2e/specs/home.spec.js [hunk 9]
```diff
@@ -2,112 +2,205 @@ const validEmail = "myCustomEmail@gmail.com";
...
+			url: "localhost:1337/game/getList",
+			body: {},
+		})
+			.its("body")
+			.then((body) => {
+				cy.get("[data-cy=game-list-item]").should(
+					"have.length",
+					body.games.length
+				);
+				console.log("body", body.games.length);
```
Comment (itsalaidbacklife) [inline tests/e2e/specs/home.spec.js:79]: stale log

File: tests/e2e/specs/home.spec.js [hunk 10]
```diff
@@ -2,112 +2,205 @@ const validEmail = "myCustomEmail@gmail.com";
...
-            
+
 			// Test that join button is now disabled
-			cy.contains('button.v-btn', 'JOIN')
-				.should('be.disabled');
+			cy.contains("button.v-btn", "JOIN").should("be.disabled");
 		});
+	});
+
+	it.only("Re-enable join when a user leaves a full lobby", () => {
```
Comment (itsalaidbacklife) [inline tests/e2e/specs/home.spec.js:141]: stale .only; must remove to run other tests

**Review by itsalaidbacklife — CHANGES_REQUESTED** [review]
Looks good. Requesting a few small changes:

Could you remove the remaining console.logs (except the ones required to keep the linter from futzing with unused request handing params, per my comments.

Last, there's still a it.only() in the home spec; pleas remove.

File: src/plugins/sails.js (hunk 1 above)
Comment (lihaobhsfer) [reply src/plugins/sails.js:2]: No particular reason. Just what I've been using with default linter options.

Comment (lihaobhsfer) [reply src/plugins/sails.js:2]: We should look into linter rules. I'll disable Prettier on my side for this project for now before it messes up with more code.

**Pull request merged**